### ISSUE TITLE
[Feature request] Disk specs as input data for the dynamic partitioning function

### ISSUE THREAD

**Issue by mechan1sm** [nix-community/disko#894]
## Problem
Here is an example partition layout:
- bootloader1
  - size = `exactly 31 MiB`
  - fs = FAT16
- root
  - size = `rest of space` && `multiple of 4GiB`
  - fs = XFS
- boot
  - size = `multiple of 16MiB` && `>= 2GiB` && `< 6GiB`
  - fs = ext4
- bootloader2
  - size = `multiple of 1MiB` && `>= 8 MiB` && `< 24 MiB`
  - fs = FAT16

afaik it is not possible to write a disko config, so that any disk will be partitioned according to the rules above,
mainly because there is no input information about the disk specs

## Possible solution
make it possible to define the `content` as the function:
```nix
disk.example = {
  device = "/dev/loop0";
  type = "disk";
  content = disk_specs : {
    ...
  };
};
```
where disk_specs is the attrset consisting of the following fields:
```
{
  sector_size: integer; (512 | 4096);
  size_in_sectors: integer;
  size_in_bytes: integer;
}
```
there might also be the actual physical block size, that disk uses internally, but I'm unsure if it's even possible to get this one

## Notes
May be related to #728 and #789

**Comment by iFreilicht** [comment]
Yes, this might be possible to achieve with #789. We can retrieve information about the sustem and pass it into the module evaluation function. We do have to make sure that it's possible to evaluate the config even without that info, though. As I understand, you only care about values related to partition sizing, so I would suggest we change the options `start`, `size` and `end` to allow for functions instead. This would limit the influence the dynamic values can have.

I'm not sure if we use then to set any NixOS options, but if we are not, we could define default empty values for the function inputs so the evaluation as a module as part of the NixOS configuration doesn't break.

---

In the meantime, you can should be able to get all these values with `lsblk -O` (I think physical block size is part of this as well) and then change the start of your config to something like


```nix
let
  sector_size = 512;
  size_in_sectors = 10000000;
  size_in_bytes = 5120000000;
in
{
```
